pyuoi/linear_model/base.py

```diff
@@ -0,0 +1,505 @@
+        n_reg_params = coefs.shape[1]
+        n_features = coefs.shape[2]
+        supports = np.zeros(
+            (n_selection_thresholds, n_reg_params, n_features),
+            dtype=bool
+        )
+
+        # iterate over each stability selection threshold
+        for thresh_idx, threshold in enumerate(selection_thresholds):
+            # calculate the support given the specific selection threshold
+            supports[thresh_idx, ...] = \
+                np.count_nonzero(coefs, axis=0) >= threshold
+
+        # unravel the dimension corresponding to selection thresholds
+        supports = np.squeeze(np.reshape(
+            supports,
+            (n_selection_thresholds * n_reg_params, n_features)
+        ))
+
+        # # TODO: collapse duplicate supports
+        return supports
+
+    @staticmethod
+    def score_predictions(metric, y_true, y_pred, metric='r2', negate=False, **kwargs):
+        """Score, according to some metric, predictions provided by a model.
+
+        the resulting score will be negated if an information criterion is
+        specified
+
+        Parameters
+        ----------
+        metric : str
+            The scoring metric to use. Acceptible options are 'AIC', 'AICc', 'BIC', and 'r2'
+
+        y_true : array-like
+            The true response variables.
+
+        y_pred : array-like
+            The predicted response variables.
+
+        metric : string
+            The type of score to run on the prediction. Valid options include
+            'r2' (explained variance), 'BIC' (Bayesian information criterion),
+            'AIC' (Akaike information criterion), and 'AICc' (corrected AIC).
+
+        negate : bool
+            Whether to negate the score. Useful in cases like AIC and BIC,
+            where minimum score is preferable.
+
+        Returns
+        -------
+        score : float
+            The score.
+        """
+
+        if metric == 'r2':
+            score = r2_score(
+                y_true=y_true,
+                y_pred=y_pred
+            )
+        elif metric == 'BIC':
+            score = utils.BIC(
+                y_true=y_true,
+                y_pred=y_pred,
+                n_features=kwargs.get('n_features')
+            )
+        elif metric == 'AIC':
+            score = utils.AIC(
+                y_true=y_true,
+                y_pred=y_pred,
+                n_features=kwargs.get('n_features')
+            )
+        elif metric == 'AICc':
+            score = utils.AICc(
+                y_true=y_true,
+                y_pred=y_pred,
+                n_features=kwargs.get('n_features')
+            )
+        else:
+            raise ValueError(
+                metric + ' is not a valid option.'
+            )
+
+        if metric in ('BIC', 'AIC', 'AICc'):
+            score = -score
+
+        return score
+
+    def uoi_selection_sweep(self, X, y, reg_param_values):
+        """Perform Lasso regression on a dataset over a sweep
```

Comment:
Not necessarily Lasso here anymore.
```suggestion
        """Perform the selection regression step on a dataset over a sweep
```